base/workerpool.jl

```diff
@@ -0,0 +1,85 @@
+# This file is a part of Julia. License is MIT: http://julialang.org/license
+
+
+type WorkerPool
+    channel::RemoteChannel{Channel{Int}}
+end
+
+
+isbusy(pool::WorkerPool) = !isready(pool.channel)
+
+
+"""
+    WorkerPool(workers)
+
+Create a WorkerPool from a vector of worker ids.
+"""
+function WorkerPool(workers::Vector{Int})
+
+    # Create a shared queue of workers...
+    pool = RemoteChannel(()->Channel{Int}(128)) 
+
+    # Check that workers are not already part of a pool...
+    check = () -> if :_worker_pool in names(Main)
+        error("Worker $(myid()) already in a WorkerPool!")
+    end
+    foreach(fetch, [@spawnat w check() for w in workers])
+
+    # Put each worker into the pool...
+    for w in workers
+        put!(pool, w)
+        @spawnat w global _worker_pool = pool
+    end
+
+    WorkerPool(pool)
+end
+
+
+"""
+    WorkerPool(n)
+
+Launches `n` workers using the in-built LocalManager.
+"""
+WorkerPool(n::Integer) = WorkerPool(addprocs(n))
+
+
+"""
+    WorkerPool()
+
+Equivalent to WorkerPool(CPU_CORES)
+"""
+WorkerPool() = WorkerPool(addprocs())
```

Comment:
`WorkerPool()` as an empty pool to which we can add workers is more appropriate? In the use case in https://github.com/JuliaLang/julia/issues/14843#issuecomment-181726648, workers are added to the pool as and when they come online. `put!(::WorkerPool, pid)` would add to the pool.